rfc/rfc006-distributed-registry.md

```diff
@@ -0,0 +1,378 @@
+* **id**: MUST contain the organization's ID encoded as URN. This can be any valid URN, for instance an AGB-code or
+  KVK-number e.g.: `urn:oid:2.16.840.1.113883.2.4.6.1:06123456` (AGB-code). It is used to refer to the organization by the
+  vendor itself or other vendors. The organisation ID SHALL NOT be changed.
+* **vid** (vendor ID): MUST be the organization's software vendor's ID as URN (e.g. `urn:oid:1.3.6.1.4.1.54851.4:1234`).
+  The vendor ID SHALL NOT be changed; relation to a care organization SHALL NOT be transfered to another vendor. 
+* **name**: MUST contain the organization's commonly known name as string. Only alphanumeric, dashes (`-`) and space characters are allowed,
+  it SHALL NOT be empty or only contain spaces. It MUST conform to the following regex: `[a-zA-Z0-9 ]+`.
+* **prfs** (proofs): MUST contain an array with cryptographic proofs asserting the organization as being an actual care organization.
+  at the same it serves as commitment from the care organization that it uses the vendor's (identified by **vid**) software.
+  See section 7 for supported proof types.
+
+Example:
+```json
+{
+    "id": "urn:oid:2.16.840.1.113883.2.4.6.1:06123456",
+    "vid": "urn:oid:1.3.6.1.4.1.54851.4:1234",
+    "name": "HomeSweetHome Thuiszorg",
+    "prfs": ["organisation proof"]
+}
+```
+
+#### Validation
+When processing an organization registration or update it MUST be validated as follows:
+
+1. Assert all required fields are present.
+2. Assert all fields value formats are valid.
+3. For updates: assert only modifiable fields are updated (**name**, **prfs** are modifiable).
+4. Assert the document is signed by the vendor identified by **vid**.
+5. Validate **prfs**:
+   * Assert there is at least one proof.
+   * Assert proofs are valid at the time they're introduced.
+   * Assert all proofs authenticate the same organisation.
+
+### 5.3. Endpoint
+This document registers an endpoint exposed by a vendor which CAN be used to exchange data with the vendor or its organizations.
+It's identified by the type `registry/endpoint`.
+
+#### Fields
+The following fields MUST be present:
+* **id**: MUST be a string identifying this endpoint. It must be unique for the vendor.
+* **vid** (vendor ID): MUST contain the ID of the vendor this endpoint belongs to.
+* **type**: MUST be a URN describing the type of this endpoint in the form of `urn:oid:1.3.6.1.4.1.54851.2:type`.
+  The actual type MUST be an ASCII alphanumeric (a-z, 0-9) lowercase string.
+* **loc** (location): MUST be a URL pointing to where the endpoint is exposed (e.g. `https://nuts.nl/some/path`)
+* **nbf** (not before): MUST contain an RFC 3339 timestamp indicating from what date/time the endpoint SHOULD be considered valid.
+
+The following fields MAY be present:
+* **exp** (expiration): MUST contain an RFC 3339 timestamp indicating when the endpoint expires and SHOULD NOT be considered valid anymore.
+
+Example:
+```json
+{
+    "id": "de59b062-d783-4727-a05e-57ed6035f00d",
```

Comment:
just plain ID or urn?

Reply:
free-format ID since it must be unique just for the vendor